### ISSUE TITLE
`EditorInterface.get_edited_scene_root().queue_free()` will crash the editor.

### ISSUE THREAD

**Issue by ryevdokimov** [godotengine/godot#90176]
### Tested versions

v4.3.dev5.mono.official [89f70e98d]

### System information

Godot v4.3.dev5.mono - Windows 10.0.22631 - Vulkan (Forward+) - integrated AMD Radeon(TM) Graphics (Advanced Micro Devices, Inc.; 31.0.24002.92) - AMD Ryzen 7 PRO 6850U with Radeon Graphics (16 Threads)

### Issue description

Might be something dumb here, but trying to delete the root node via the title method to go back to this in the editor:

![image](https://github.com/godotengine/godot/assets/105675984/66ae63ad-9fbd-4fc7-8b7d-e6dcc0ead0e9)

Get a crash instead. 

ERROR: Cannot get path of node as it is not in a scene tree.
   at: (scene/main/node.cpp:2188)
ERROR: Condition "!is_inside_tree()" is true. Returning: false
   at: can_process (scene/main/node.cpp:791)

### Steps to reproduce

File->Run

```@tool
extends EditorScript

func _run() -> void:
	EditorInterface.get_edited_scene_root().queue_free()
```

### Minimal reproduction project (MRP)

N/A

**Comment by AThousandShips** [comment]
The best we could do here I'd say is prevent it being freed, otherwise this is a case in my opinion if destructive use of the engine, I'd say you should expect issues if you try to delete stuff that the editor handles

**Comment by ryevdokimov** [comment]
Maybe.

Deleting the root node isn't outside normal usage of the engine. 

![image](https://github.com/godotengine/godot/assets/105675984/a2cf9ade-9cb4-422b-9747-6c484ecbfdb5)

Doing this will work and take you to the screenshot in the issue. I'm basically looking for a reverse of this (if/when it gets implemented): https://github.com/godotengine/godot/pull/90136, to build out an undo/redo system for plugins.

**Comment by ryevdokimov** [comment]
Well yeah, it crashes the editor lol. I'll agree with that point. It would be nice if I _could_ delete the root node via scripts, but at a minimum, sure it shouldn't crash.

**Comment by KoBeWi** [comment]
Related: #26409

**Comment by AThousandShips** [comment]
So what would the solution be? To just complain and close the tab without saving I assume? There's no way to recover the node when deleted

**Comment by ryevdokimov** [comment]
I don't see why not. You also don't have a way to recover nodes you delete when you use `remove_child()`, nor does it prompt you either. At least not without programming in that functionality.

**Comment by AThousandShips** [comment]
Then the solution is to just recover gracefully, as opposed to crashing, and not trying to fix anything, that's a good solution, but people might not like it (also `remove_child` doesn't delete anything, so not really comparable)

**Comment by ryevdokimov** [comment]
Fair point, I suppose `queue_free()` on children is comparable and the result looks the same.
